### PR TITLE
[16.04] fix image generation for workflows with missing tools

### PR DESCRIPTION
**Pull request by martenson**
currently this ends up with a server error

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/controllers/workflow.py [hunk 1]
```diff
@@ -512,8 +512,14 @@ def get_name_and_link_async( self, trans, id=None ):
     @web.require_login( "use Galaxy workflows" )
     def gen_image( self, trans, id ):
         stored = self.get_stored_workflow( trans, id, check_ownership=True )
+        try:
+            svg = self._workflow_to_svg_canvas( trans, stored )
+        except Exception:
+            status = 'error'
+            message = 'Galaxy is unable to create the svg image. Please check your workflow, there might be missing tools.'
```
Comment (bgruening) [inline lib/galaxy/webapps/galaxy/controllers/workflow.py]: SVG

**Comment by bgruening** [comment]
:+1:

File: lib/galaxy/webapps/galaxy/controllers/workflow.py [hunk 2]
```diff
@@ -1282,6 +1288,8 @@ def _workflow_to_svg_canvas( self, trans, stored ):
         for step in workflow.steps:
             # Load from database representation
             module = module_factory.from_workflow_step( trans, step )
+            if not module:
+                raise Exception
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/controllers/workflow.py]: I think an Exception would be raised anyway on the next line.

Comment (martenson) [inline lib/galaxy/webapps/galaxy/controllers/workflow.py]: good point, dropped

**Comment by nsoranzo** [comment]
Thanks @martenson!

**Pull request merged**